### ISSUE TITLE
Support preview Team Discussions API

### ISSUE THREAD

**Issue by gmlewis** [google/go-github#849]
GitHub Developer API announcement:
https://developer.github.com/changes/2018-02-07-team-discussions-api/

This would be a great PR for any new contributor to this repo or a new Go developer.
All contributions are greatly appreciated!

Feel free to volunteer for any issue, and we can send you an invite to contribute to the repo (which you then accept) and the issue can be assigned to you so that others don't attempt to duplicate the work.

Make sure you have two-factor authentication enabled on GitHub... as we will be requiring it soon to be a contributor. It's a good idea anyway... in general... not just for GitHub!  :smile: 

Thank you!

**Comment by parkhyukjun89** [comment]
Hi @gmlewis, can I work on this issue ?

**Comment by parkhyukjun89** [comment]
@gmlewis Accepted!
One quick question. I noticed that team service codes (for https://developer.github.com/v3/teams/members/) still lives under orgs_teams.go.
Would it make sense to create new files (teams_discussions.go, teams_discussion_comments.go) for these changes or should I just use the same orgs_teams.go ?

**Comment by gmlewis** [comment]
Our code organization attempts to closely follow the layout of the GitHub Developer documentation.

Since their URL starts with `/teams/`, I think it is highly appropriate to create new `teams_*.go` files as you suggested.

If it is small enough to all go into `teams_members.go`, then I would say this would be optimal. But if you think that it is getting too large (use the other files in the repo for comparison), then you could split it out into separate files. Of course, this can also be discussed with the greater collection of GitHub Authors in the PR as we go along if anyone has strong opinions.  :smile:

**Comment by gmlewis** [comment]
Excellent catch, @parkhyukjun89!

Yes, we would like to keep our code hierarchy closely matching the GitHub Developer API documentation hierarchy, so moving it makes sense to me based on that announcement.

**Comment by gmlewis** [comment]
Since the discussion was started here, reopening issue to track the changes in #904.

**Comment by jamesgeorge007** [comment]
@gmlewis What is to be done further here?

**Comment by gmlewis** [comment]
This issue needs someone to investigate the full Team Discussions API on the GitHub Developer v3 API site, then study the code in this repo, and either report back that everything is supported and up-to-date in the repo (and this issue can be closed), or more work needs to be done. If more work needs to be done, the owner of this issue will ideally create a PR that addresses the remaining API endpoints that need support.

So this issue, like most issues on this repo (and possibly most repos in general) involve volunteers  digging into the existing documentation and code and figuring out what needs to be done, and then doing it.

I apologize if this sounds harsh... that is not my intent.

As I pointed out in [another PR](https://github.com/google/go-github/pull/1102#issuecomment-457589767), this repo attracts many new-to-Go developers who want to learn Go and contribute to an obviously-successful project... and this is a great thing! We absolutely appreciate [all our volunteers](https://github.com/google/go-github/blob/master/AUTHORS) and the excellent work that they contribute!

Part of that task is figuring out what needs to be done.

One of the maintainers could go through periodically and spec out all the work required in every issue, but that would not only be a monumental task, but would rob others from the joy and learning experience provided by doing this work themselves. 😄 It also adds to the satisfaction of a job well-done.

Thank you in advance for your contributions to this repo, and I look forward to seeing discussions and PRs that you contribute.

**Comment by parkhyukjun89** [comment]
Sorry. I haven't been following up with this issue after I created PRs. Bascially, there were 2 PRs to this issue, one addressing new APIs added to Teams API (https://github.com/google/go-github/pull/889) and one other addressing updates to Teams API (https://github.com/google/go-github/pull/904/files, which later was merged as a part of https://github.com/google/go-github/pull/972 due to merge conflicts). Since both were merged while back, I believe this issue can be closed, unless, of course, there were additional updates to Teams API, which may need to be tracked in a separate issue.
